tasks/mrbgems.rake

```diff
@@ -34,20 +31,42 @@ MRuby.each_target do
         f.puts %Q[ */]
         f.puts %Q[]
         f.puts %Q[#include <mruby.h>]
+        f.puts %Q[#include <mruby/proc.h>]
         f.puts %Q[]
         f.write gem_func_decls
-        unless gem_final_calls.empty?
-          f.puts %Q[]
-          f.puts %Q[static void]
-          f.puts %Q[mrb_final_mrbgems(mrb_state *mrb) {]
-          f.write gem_final_calls
-          f.puts %Q[}]
-        end
+        f.puts %Q[]
+        f.puts %Q[static const struct {]
+        f.puts %Q[  void (*init)(mrb_state *);]
+        f.puts %Q[  void (*final)(mrb_state *);]
+        f.puts %Q[} gem_funcs[] = {]
```

Comment:
Do we need to pack init/finish functions in an array? What is the benefit?

Reply:
Without the function array, the function would be expanded directly, and error handling would have to be written each time as well.

Reply:
The final form is the same, but I would change the form at the commit level.

Reply:
Changes have been completed.
I also changed the commit message.